src/trunk.c

```diff
@@ -8182,6 +8199,61 @@ trunk_print(platform_log_handle *log_handle, trunk_handle *spl)
    trunk_print_subtree(log_handle, spl, spl->root_addr);
 }
 
+/* Print meta-page's linked list for one routing filter at address 'meta_head'.
+ */
+void
+trunk_print_filter_metapage_list(platform_log_handle *log_handle,
+                                 trunk_handle        *spl,
+                                 uint64               meta_head)
+{
+   platform_log(log_handle,
+                "\nFilter Metadata page starting from meta_head=%lu\n{\n",
+                meta_head);
+   mini_unkeyed_print(spl->cc, meta_head, PAGE_TYPE_FILTER);
```

Comment:
The code executes just fine all the way through when we make the call from `splinter_test.c:test_splinter_print_diags()` to the newly added driver function, `trunk_print_root_nodes_filter_metapages()`.

But the output comes out as garbage. For example see this:

```
** Trunk nodes filter metapages list: trunk_print_root_nodes_filter_metapages() **

Filter Metadata page starting from meta_head=6200492032
{
---------------------------------------------
| Mini Allocator -- meta_head:   6200492032 |
|-------------------------------------------|
| idx |                         extent_addr |
|-------------------------------------------|
| meta addr=6200492032      , num_entries=1905657808      |
|-------------------------------------------|
|   0 |                 8184746964978499585 |
|   1 |                 8184751560593506305 |
|   2 |                 8184756237812891649 |
[...]
```

The `num_entries` value printed as `num_entries=1905657808` seems bogus.

So the following `extent_addr` for each entry also seems bogus.

Did the routing filter's metadata page not ever get built, first of all? Are we accessing an uninitialized page?

Reply:
The print method will run forever (as `num_entries` value is too huge) or will seg-fault. The latter  usually is what happens with debug builds, which is why CI-runs are failing miserably.